packages/x-date-pickers/src/TimeClock/Clock.tsx

```diff
@@ -145,56 +145,41 @@ const ClockPin = styled('div', {
   transform: 'translate(-50%, -50%)',
 }));
 
+const meridiemButtonCommonStyles = (theme: Theme) => ({
+  zIndex: 1,
+  bottom: 8,
+  paddingLeft: 4,
+  paddingRight: 4,
+  width: CLOCK_HOUR_WIDTH,
+  [`&.${clockClasses.selected}`]: {
```

Comment:
Does this introduce any specificity change?

Reply:
Great point.
This could be a BC for someone currently overriding selected styles via theme.
I'll check.
As for the class specificity, I don't think we can run into it, because there was no way to identify the selected button based on DOM/Classes for now. 🙈

Reply:
If people are using the variants / the `ownerState` in their theme, would they see their CSS change because of specificity change on our side?
I think so but I'm not very good at CSS specificity :laughing:

Reply:
I've double-checked and indeed, we can't go this route, at least now, because for someone it would be a BC. 🙈 
The following styles would no longer apply to a selected meridiem button:
```ts
MuiClock: {
  styleOverrides: {
    amButton: {
      color: 'red',
    },
    pmButton: {
      color: 'green',
    },
  },
},
```

Reply:
If you think we should do the change in the next major, feel free to open an issue so that we can keep track of it :+1: